src/main/java/za/co/neslotech/spring/trainspringbootinsight/auth/filter/BasicAuthenticationFilter.java

```diff
@@ -0,0 +1,77 @@
+package za.co.neslotech.spring.trainspringbootinsight.auth.filter;
+
+import io.jsonwebtoken.io.Decoders;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.lang.NonNull;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import za.co.neslotech.spring.trainspringbootinsight.service.BasicAuthenticationService;
+
+import java.io.IOException;
+
+@Component
+public class BasicAuthenticationFilter extends OncePerRequestFilter {
+
+    public static final String HEADER_AUTHORIZATION = "Authorization";
+    public static final String AUTHORIZATION_BASIC = "Basic";
+    public static final String SEPARATOR = ":";
+    private final BasicAuthenticationService basicAuthService;
+    private final UserDetailsService userDetailsService;
+
+    public BasicAuthenticationFilter(final BasicAuthenticationService basicAuthService,
+                                     final UserDetailsService userDetailsService) {
+        this.basicAuthService = basicAuthService;
+        this.userDetailsService = userDetailsService;
+    }
+
+    @Override
+    protected void doFilterInternal(@NonNull final HttpServletRequest request,
+                                    @NonNull final HttpServletResponse response,
+                                    final FilterChain filterChain)
+    throws ServletException, IOException {
+
+        processBasicAuth(request);
+        filterChain.doFilter(request, response);
+    }
+
+    private void processBasicAuth(final HttpServletRequest request) {
+
+        final var authHeader = request.getHeader(HEADER_AUTHORIZATION);
+
+        if (authHeader != null && authHeader.startsWith(AUTHORIZATION_BASIC)) {
```

Comment:
You can escape this function early and throw an exception if the auth header is null.

Reply:
Also what would be the case that the auth header does not start with `Basic `?